client.go

```diff
@@ -603,6 +615,12 @@ func (c *Client) execute(req *Request) (*Response, error) {
 
 	req.Time = time.Now()
 	c.httpClient.Transport = c.transport
+
+	if req.RawRequest.URL != nil && req.RawRequest.URL.Scheme == "" {
```

Comment:
Can you please move this complete `if` condition to [middleware.go#L141](https://github.com/go-resty/resty/blob/master/middleware.go#L141)?